### PR TITLE
Flink History Server kerberos enabled

### PR DESCRIPTION
**Pull request by wForget**

### CODE REVIEW

File: templates/hadoop-master/files/etc/flink/conf_history_server/flink-conf.yaml [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+# 4. configure the connector to use JAAS/SASL
+
+# The below configure how Kerberos credentials are provided. A keytab will be used instead of
+# a ticket cache if the keytab path and principal are set.
+
+# security.kerberos.login.use-ticket-cache: true
+# security.kerberos.login.keytab: /path/to/kerberos/keytab
+# security.kerberos.login.principal: flink-user
+
+{% if kerberos_enabled %}
```
Comment (wForget) [inline templates/hadoop-master/files/etc/flink/conf_history_server/flink-conf.yaml:224]: added secuity configs

File: templates/hadoop-master/files/etc/supervisor/conf.d/mapred-history-server.conf [hunk 2]
```diff
@@ -1,4 +1,4 @@
-[program:mapred_history_server]
+[program:mapred-history-server]
```
Comment (wForget) [inline templates/hadoop-master/files/etc/supervisor/conf.d/mapred-history-server.conf:1]: irrelevant change, consistent naming style

File: templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf [hunk 3]
```diff
@@ -10,4 +10,5 @@ autostart=true
 
 environment=
     JAVA_HOME=/opt/openjdk-8,
+    FLINK_CONF_DIR=/etc/flink/conf_history_server,
```
Comment (pan3793) [inline templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf:13]: is it possible to share the FLINK_CONF_DIR with other Flink components?

Comment (wForget) [reply templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf:13]: Flink cannot set separate keytab and principal for history server, which means that the kerberos configurations we added applies to all flink jobs.

Comment (wForget) [reply templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf:13]: For other jobs, we can use tgt cache without additional configurations.

Comment (pan3793) [reply templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf:13]: > Flink cannot set separate keytab and principal for history server

an awkward design ...

Comment (pan3793) [reply templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf:13]: can we minimize the configuration to keep only history server related items?

Comment (pan3793) [reply templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf:13]: The Flink community somedays may add the HS dedicated kerberos configuration items then we can merge them back.

Comment (wForget) [reply templates/hadoop-master/files/etc/supervisor/conf.d/flink-history-server.conf:13]: Makes sense, I'll try it later.

File: templates/hadoop-master/Dockerfile.j2 [hunk 4]
```diff
@@ -46,6 +46,7 @@ ENV SPARK_CONF_DIR=/etc/spark/conf
 {% if flink_enabled %}
 ENV FLINK_HOME=/opt/flink
 ENV FLINK_CONF_DIR=/etc/flink/conf
+ENV FLINK_LOG_DIR=/var/log/flink
```
Comment (wForget) [inline templates/hadoop-master/Dockerfile.j2:49]: set flink_log_dir

File: templates/hadoop-master/files/etc/flink/conf_history_server/log4j-console.properties [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+# Log all infos to the console
+appender.console.name = ConsoleAppender
+appender.console.type = CONSOLE
+appender.console.layout.type = PatternLayout
+appender.console.layout.pattern = %d{yyyy-MM-dd HH:mm:ss,SSS} %-5p %-60c %x - %m%n
+appender.console.filter.threshold.type = ThresholdFilter
+appender.console.filter.threshold.level = ${sys:console.log.level:-ALL}
+
+# Log all infos in the given rolling file
+appender.rolling.name = RollingFileAppender
```
Comment (pan3793) [inline templates/hadoop-master/files/etc/flink/conf_history_server/log4j-console.properties:54]: remove file related parts, we just need to log everything to stdout/stderr, and supervisord handles that.

Comment (wForget) [reply templates/hadoop-master/files/etc/flink/conf_history_server/log4j-console.properties]: thanks, fixed

**Review by pan3793 — APPROVED** [review]

**Pull request merged**